File: fortran/04-control-flow/loop_control.f90
program loop_control
    implicit none
    integer :: i, j

    ! CYCLE skips to the next iteration (like "continue")
    do i = 1, 6
        if (mod(i, 2) == 0) cycle   ! skip even numbers
        write(*, '(A, I0)') "Odd: ", i
    end do

    ! EXIT leaves the loop early (like "break")
    do i = 1, 100
        if (i > 3) exit
        write(*, '(A, I0)') "Counting: ", i
    end do

    ! Named loops let EXIT target a specific outer loop
    outer: do i = 1, 3
        inner: do j = 1, 3
            if (i + j == 4) exit outer
            write(*, '(A, I0, A, I0)') "i=", i, " j=", j
        end do inner
    end do outer
end program loop_control

File: javascript/04-control-flow/loop_control.js
// Controlling loop execution

// break exits the loop entirely
for (let i = 1; i <= 10; i++) {
  if (i > 5) {
    break;
  }
  console.log(`break demo: ${i}`);
}

// continue skips to the next iteration
for (let i = 1; i <= 6; i++) {
  if (i % 2 === 0) {
    continue; // skip even numbers
  }
  console.log(`odd number: ${i}`);
}

// A label lets break target an outer loop
outer: for (let row = 1; row <= 3; row++) {
  for (let col = 1; col <= 3; col++) {
    if (row + col === 4) {
      console.log(`stopping at row ${row}, col ${col}`);
      break outer;
    }
  }
}
